autogen/oai/client.py

```diff
@@ -408,6 +410,36 @@ def cost(self, response: Union[ChatCompletion, Completion]) -> float:
             return (tmp_price1K[0] * n_input_tokens + tmp_price1K[1] * n_output_tokens) / 1000  # type: ignore [no-any-return]
         return tmp_price1K * (n_input_tokens + n_output_tokens) / 1000  # type: ignore [operator]
 
+    def map_params(self, params: Dict[str, Any]) -> Dict[str, Any]:
+        """Maps parameters that are deprecated"""
+
+        # max_tokens is deprecated and replaced by max_completion_tokens as of 2024.09.12
+        if "max_tokens" in params:
```

Comment:
`max_completion_tokens` is for o1 models only

Reply:
I believe `max_completion_tokens` has replaced `max_tokens` and in their API documentation they note `This value is now deprecated in favor of max_completion_tokens, and is not compatible with [o1 series models](https://platform.openai.com/docs/guides/reasoning).`. 

As per OpenAI's documentation they seemed to have replaced it.
![image](https://github.com/user-attachments/assets/09a8706a-7478-456c-95c1-79699f3921e0)

[API reference](https://platform.openai.com/docs/api-reference/chat/create#chat-create-max_tokens)
![image](https://github.com/user-attachments/assets/23fba4bb-14d8-4c66-b5a1-ab4a65f40aa0)

Reply:
Let me know if you want further changes on this or we can mark it as resolved.

Reply:
With updated openai package, this should resolve now.